apps/console/src/features/identity-providers/components/forms/authenticators/saml-authenticator-settings-form.tsx

```diff
@@ -0,0 +1,814 @@
+    Form as ReactFinalForm,
+    FormRenderProps,
+    FormSpy,
+    FormSpyRenderProps
+} from "react-final-form";
+import { OnChange } from "react-final-form-listeners";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+import { Form as SemanticUIForm, Grid, SemanticWIDTHS } from "semantic-ui-react";
+import { AppState, ConfigReducerStateInterface } from "../../../../core";
+import {
+    CommonAuthenticatorFormInitialValuesInterface,
+    FederatedAuthenticatorWithMetaInterface
+} from "../../../models";
+import {
+    castToBool,
+    composeValidators,
+    fastSearch,
+    getAvailableNameIDFormats,
+    getAvailableProtocolBindingTypes,
+    getDigestAlgorithmOptionsMapped,
+    getSignatureAlgorithmOptionsMapped,
+    hasLength,
+    IDENTITY_PROVIDER_ENTITY_ID_LENGTH,
+    isUrl,
+    LOGOUT_URL_LENGTH,
+    required,
+    SERVICE_PROVIDER_ENTITY_ID_LENGTH,
+    SSO_URL_LENGTH
+} from "../../utils/saml-idp-utils";
+
+/**
+ * SamlSettingsForm Properties interface. The data-testid is added in
+ * {@link SamlAuthenticatorSettingsForm.defaultProps}.
+ */
+interface SamlSettingsFormProps extends TestableComponentInterface {
+    /**
+     * Authenticator instance. This object contains both the initial values
+     * for the form and the required metadata for form fields.
+     */
+    authenticator: FederatedAuthenticatorWithMetaInterface;
+    /**
+     * Callback for form submit.
+     * @param {CommonAuthenticatorFormInitialValuesInterface} values - Resolved Form Values.
+     */
+    onSubmit: (values: CommonAuthenticatorFormInitialValuesInterface) => void;
+}
+
+export interface SamlProperties {
+    // These are the properties we have ignored.
+    // Keeping them as a reference for future implementations.
+    //
+    // ACSUrl: string;
+    // AttributeConsumingServiceIndex: string;
+    // AuthnContextClassRef: string;
+    // AuthnContextComparisonLevel: string;
+    // ForceAuthentication: string;
+    // ISArtifactBindingEnabled: boolean;
+    // IncludeAuthnContext: string;
+    // IncludeCert: boolean;
+    // IncludeNameIDPolicy: boolean;
+    // IsAssertionEncrypted: boolean;
+    // SelectMode: string;
+    // ResponseAuthnContextClassRef: string;
+    // isAssertionSigned: boolean;
+    // meta_data_saml: string;
+    // SignatureAlgorithmPost: string;
+    // CustomAuthnContextClassRef: string;
+    DigestAlgorithm?: string;
+    ISAuthnReqSigned?: boolean;
+    IdPEntityId?: string;
+    IncludeProtocolBinding?: boolean;
+    IsAuthnRespSigned?: boolean;
+    IsLogoutEnabled?: boolean;
+    IsLogoutReqSigned?: boolean;
+    IsSLORequestAccepted?: boolean;
+    IsUserIdInClaims?: boolean;
+    LogoutReqUrl?: string;
+    NameIDType?: string;
+    RequestMethod?: string;
+    SPEntityId?: string;
+    SSOUrl?: string;
+    SignatureAlgorithm?: string;
+    commonAuthQueryParams?: string;
+}
+
+export const SamlAuthenticatorSettingsForm: FC<SamlSettingsFormProps> = (
```

Comment:
Lets use the long form to maintain consistency. Check other places as well.
```suggestion
export const SamlAuthenticatorSettingsForm: FunctionComponent<SamlSettingsFormProps> = (
```

Reply:
Addressed this with commit 051b827